### PR TITLE
APPS-1166 fix deterministic for checksum

### PR DESCRIPTION
**Pull request by YuxinShi0423**

### CODE REVIEW

File: common/src/main/scala/dx/util/JsUtils.scala [hunk 1]
```diff
@@ -220,18 +220,21 @@ object JsUtils {
...
     jsValue match {
       case JsObject(m: Map[String, JsValue]) =>
         // deterministically sort maps by using a tree-map instead
         // a hash-map
         val mTree = m
-          .map { case (k, v) => k -> JsUtils.makeDeterministic(v) }
+          .map {
+            case (k, JsArray(a)) =>
+              k -> JsArray(a.map(JsUtils.makeDeterministic).sortWith(_.toString < _.toString))
+            case (k, v) => k -> JsUtils.makeDeterministic(v)
```
Comment (commandlinegirl) [inline common/src/main/scala/dx/util/JsUtils.scala:233]: This LGTM.

I'm wondering if we need a recursive check, for cases when an array is embedded deeper? E.g. (k, v), where v is another map (q, r), where r is a map, etc...

**Review by commandlinegirl — DISMISSED** [review]

File: common/src/main/scala/dx/util/JsUtils.scala (hunk 1 above)
Comment (YuxinShi0423) [reply common/src/main/scala/dx/util/JsUtils.scala:233]: Added. Thanks for suggesting it :D

File: common/src/test/scala/dx/util/JsUtilsTest.scala [hunk 2]
```diff
@@ -14,4 +14,28 @@ class JsUtilsTest extends AnyFlatSpec with Matchers {
...
+    JsUtils.makeDeterministic(x) should be(JsUtils.makeDeterministic(y))
+
+    //nested JsObject and JsArray with order changed
+    val x2 = JsObject("a" -> JsArray(JsObject("aa" -> x, "ab" -> JsString("aba")),
+                                     JsObject("aa" -> y, "ab" -> JsString("abb"))),
+                      "b" -> y)
+    val y2 = JsObject("b" -> x,
+                      "a" -> JsArray(JsObject("ab" -> JsString("abb"), "aa" -> y),
+                                     JsObject("aa" -> x, "ab" -> JsString("aba"))))
+    JsUtils.makeDeterministic(x2) should be(JsUtils.makeDeterministic(y2))
```
Comment (YuxinShi0423) [inline common/src/test/scala/dx/util/JsUtilsTest.scala:39]: Here is the test for nested JSON. @commandlinegirl

File: common/src/main/scala/dx/util/JsUtils.scala (hunk 1 above)
Comment (commandlinegirl) [reply common/src/main/scala/dx/util/JsUtils.scala:233]: Great, thank you @YuxinShi0423!

**Review by commandlinegirl — APPROVED** [review]
LGTM, many thanks for finding and fixing this!

**Pull request merged**